src/ExceptionTransformer.ts

```diff
@@ -1,34 +1,56 @@
 import { createMapFromObject } from "./utils/mapUtils";
+import {
+  getErrorDetail,
+  isObjectEmpty,
+  getStringMessage,
+  deleteProperty,
+  getValueFromPath,
+  isArrayOfString
+} from "./utils/errorUtils";
 import {
   CustomTransformers,
   Exception,
-  ExceptionMap
+  ExceptionMap,
+  Options,
+  ExceptionDetail
 } from "./ExceptionTransformerModel";
 
+interface ExceptionTransformerConfig {
+  customTransformers?: CustomTransformers;
+  onUnexpectedException?: (details: {
+    error: any;
+    errorInfo: Exception;
+  }) => void;
+}
+
 class ExceptionTransformer {
   private readonly customTransformers?: CustomTransformers;
+  private readonly genericErrorMessage: string;
+  private readonly onUnexpectedException?: (details: {
+    type: string;
+    error: any;
+    errorInfo: Exception;
+  }) => void;
 
-  // Initialize the custom exception transformers to the instance
-  // If you need to handle custom logic according to exception.type, here is an example;
-  // const customTransformers = {
-  //   ProfileCredentialError: (exception: Exception) => {
-  //     let exceptionMap = new Map();
-  //
-  //     exceptionMap = exceptionMap.set("email", "You shall not pass.");
-  //
-  //     return exceptionMap;
-  //   },
-  //   ...
-  // };
-  public constructor(customTransformers?: CustomTransformers) {
-    if (customTransformers) {
-      this.customTransformers = customTransformers;
+  constructor(
+    genericErrorMessage: string,
+    config?: ExceptionTransformerConfig
+  ) {
+    if (config) {
+      if (config.customTransformers) {
+        this.customTransformers = config.customTransformers;
+      }
+
+      if (config.onUnexpectedException) {
+        this.onUnexpectedException = config.onUnexpectedException;
+      }
     }
+    this.genericErrorMessage = genericErrorMessage;
```

Comment:
I think it would be useful for some certain cases, such as changing language of an app, that I am able to modify genericErrorMessage after initialization

Reply:
Added a new function `changeGenericErrorMessage` to change `genericErrorMessage`